rotkehlchen/exchanges/htx.py

```diff
@@ -0,0 +1,191 @@
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import AssetWithOracles
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+HUOBI_BASE_URL: Final = 'https://api.huobi.pro'
+VALID_ACCOUNT_BALANCE_TYPES: Final = ('spot', 'otc', 'point')
+VALID_BALANCE_TYPES: Final = ('trade', 'frozen')
+
+
+class Htx(ExchangeInterface):
+    def __init__(
+            self,
+            name: str,
+            api_key: ApiKey,
+            secret: ApiSecret,
+            database: 'DBHandler',
+            msg_aggregator: 'MessagesAggregator',
+    ):
+        super().__init__(
+            name=name,
+            location=Location.HTX,
+            api_key=api_key,
+            secret=secret,
+            database=database,
+        )
+        self.valid_account_balances_types = VALID_ACCOUNT_BALANCE_TYPES
+        self.valid_balance_types = VALID_BALANCE_TYPES
+        self.msg_aggregator = msg_aggregator
+
+    def validate_api_key(self) -> tuple[bool, str]:
+        """Validates that the HTX API key is good for usage in rotki"""
+        try:
+            accounts = self.get_accounts()
+        except RemoteError as e:
+            return False, str(e)
+
+        if len(accounts) != 0:
+            return True, ''
+
+        return False, 'API key cannot access account information'
+
+    def _sign_request(self, url: str, params: dict[str, Any]) -> dict[str, Any]:
+        """
+        Create signature for the provided payload.
+        https://www.htx.com/en-us/opend/newApiPages/?id=419
+        """
+        new_params = params.copy()  # copy to avoid modifications in params if they are used later
+        new_params['AccessKeyId'] = self.api_key
+        new_params['SignatureVersion'] = '2'
+        new_params['SignatureMethod'] = 'HmacSHA256'
+        new_params['Timestamp'] = datetime.datetime.now(datetime.UTC).strftime('%Y-%m-%dT%H:%M:%S')
+
+        host = urllib.parse.urlparse(url).hostname
+        path = urllib.parse.urlparse(url).path
+        param_str = '&'.join(  # params need to be sorted to be correctly validated
+            [
+                str(k) + '=' + urllib.parse.quote_plus(str(v))  # need to use the quoted string since cursors have `=` and it breaks signatures  # noqa: E501
+                for k, v in sorted(new_params.items())
+                if v is not None
+            ],
+        )
+        # api describes that strings should be joined using '\n'
+        payload = f'GET\n{host}\n{path}\n{param_str}'
+        dig = hmac.new(
+            key=self.secret,
+            msg=payload.encode('utf-8'),
+            digestmod=hashlib.sha256,
+        ).digest()
+        new_params['Signature'] = base64.b64encode(dig).decode()
+        return new_params
+
+    def _query(self, absolute_path: str) -> Any:
+        url = f'{HUOBI_BASE_URL}{absolute_path}'
```

Comment:
no need for this variable. If it's used only in one place as I already said in last review